.github/workflows/labeler.yml

```
@@ -0,0 +1,14 @@
+name: "Labeler"
+on:
+  pull_request:
+    types: [opened, edited]
+
```

I believe these are the permissions needed for this action to work correctly. Note that if an external contributor somehow gets access to the `secrets.GITHUB_TOKEN` they could (at most) change all our labels before it expires 
```suggestion

permissions:
  issues: write
  pull-requests: read

```

---

Seems that `triage` is the minimum for applying/dismissing labels:
https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role

Maybe we should use `triage` for both?

---

Unfortunately GHAs use a different permission naming scheme: https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs 

So only `read`, `write`, and `none` are our options. They default to `none` for `pull_request_target`. I also don't think this workflow explicitly needs `pull-requests: read` because it uses the `context` to get the pull request information. We should remove it. 

I double-checked the action source at https://github.com/github/issue-labeler/blob/master/src/main.ts and I believe we actually need `contents: read` so it can read the config file.

---

```suggestion
permissions:
  issues: write
  contents: read

```

---

Didn't expect the permission set discrepancy at all lol. Appreciated.

For anyone interested, here are the lists of API calls that the [`pull-requests`](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps#pull-requests) and [`contents`](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps#contents) permissions open up respectively.